File: cses/leetcode/python/2363.py
from collections import defaultdict
class Solution:
    def mergeSimilarItems(self, items1: List[List[int]], items2: List[List[int]]) -> List[List[int]]:
        mp = defaultdict(int)
        for item, weight in items1:
            mp[item]+=weight
        for item, weight in items2:
            mp[item]+=weight
        ans = []
        for item,weight in mp.items():
            ans.append([item,weight])
        ans = sorted(ans)
        return ans

File: leetcode/cpp/2363.cpp

#include <bits/stdc++.h>
using namespace std;
class Solution
{
public:
    vector<vector<int>> mergeSimilarItems(vector<vector<int>> &items1, vector<vector<int>> &items2)
    {
        map<int, int> mp; // value to weight
        for (vector<int> p : items1)
        {
            if (mp[p[0]] == 0)
            {
                mp[p[0]] = p[1];
            }
            else
            {
                mp[p[0]] += p[1];
            }
        }
        for (vector<int> p : items2)
        {
            if (mp[p[0]] == 0)
            {
                mp[p[0]] = p[1];
            }
            else
            {
                mp[p[0]] += p[1];
            }
        }
        vector<vector<int>> ans;
        for (auto it : mp)
        {
            vector<int> v;
            v.push_back(it.first);
            v.push_back(it.second);
            ans.push_back(v);
        }
        return ans;
    }
};
int main()
{
}